test/mpi/impls/mpich/mpi_t/recvq_events.c

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright (C) by Argonne National Laboratory
+ *     See COPYRIGHT in top-level directory
+ */
+
+/* A simple test of the proposed MPI_T_ interface that assumes the
+ * presence of two events: "unexp_message_enqueued" and
+ * "unexp_message_dequeued".  Some information about these events
+ * is printed to stdout and then a few messages are sent/received to
+ * show that the call backs for these events are triggered.
+ *
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <assert.h>
+#include <string.h>
+#include "mpi.h"
+#include "mpitestconf.h"
+#include "mpitest.h"
+
+int errs = 0;
+
+void uenq_cb_fn(MPI_T_event_instance event_instance, MPI_T_event_registration er,
+                MPI_T_cb_safety cb_safety, void *user_data)
+{
+    int rank;
+    int tag;
+    MPI_T_event_read(event_instance, 0, &rank);
+    MPI_T_event_read(event_instance, 1, &tag);
+    if (rank != 0 || tag != 11)
+        errs = 1;
+}
+
+void udeq_cb_fn(MPI_T_event_instance event_instance, MPI_T_event_registration er,
+                MPI_T_cb_safety cb_safety, void *user_data)
+{
+    int rank;
+    int tag;
+    MPI_T_event_read(event_instance, 0, &rank);
+    MPI_T_event_read(event_instance, 1, &tag);
+    if (rank != 0 || tag != 11)
+        errs = 1;
+}
+
+void recvq_free_cb(MPI_T_event_registration event_registration, MPI_T_cb_safety cb_safety,
+                   void *user_data)
+{
+    /* */
+}
+
+
+int main(int argc, char *argv[])
+{
+    int i;
+    int num;
+    int rank, size;
+#define STR_SZ (50)
+    char *name;
+    char *desc;
+    int verbos;
+    MPI_Datatype *dt;
+    MPI_Aint *at;
+    MPI_Datatype dtype;
+    int count, num_elements;
```

Comment:
Please check for unused variables in the test program and remove them. If MPICH is configured with `--enable-strict`, it should be able to flag them for you.